lib/redispub/publisher.go

```diff
@@ -63,26 +63,33 @@ var metricLastCommandDuration = promauto.NewGauge(prometheus.GaugeOpts{
 
 // PublishStream reads Publications from the given channel and publishes them
 // to Redis.
-func PublishStream(clients []redis.UniversalClient, in <-chan *Publication, opts *PublishOpts, stop <-chan bool) {
+func PublishStream(clients []redis.UniversalClient, in <-chan *Publication, opts *PublishOpts, stop <-chan bool, ordinal int) {
+	metricLastOplogEntryStaleness := promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace: "otr",
+		Subsystem: "redispub",
+		Name:      fmt.Sprintf("last_entry_staleness_seconds_%d", ordinal),
```

Comment:
I would make the ordinal a label rather than interpolating it into the metric name